### PR TITLE
feat: add svelte-check to lint process

### PR DESCRIPTION
**Pull request by rob-3**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "eslint-config-custom": "file:*",
     "prettier": "3.0.3",
     "prettier-config-custom": "file:*",
+    "svelte-check": "^3.6.0",
```
Comment (rikimbili) [inline package.json:17]: Let's use the exact version

File: repos/web/package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "preview": "vite preview",
     "check": "svelte-check --tsconfig ./tsconfig.json",
     "check:watch": "svelte-check --tsconfig ./tsconfig.json --watch",
-    "lint": "prettier --check --ignore-path=../../.prettierignore . && eslint \"src\"",
+    "lint": "prettier --check --ignore-path=../../.prettierignore . && eslint \"src\" && svelte-check",
```
Comment (rikimbili) [inline repos/web/package.json:11]: From what I understand, ESLint already does a lot of things that svelte-check does. But I’m not sure if it covers everything. Can you help me understand the differences between the two in terms of what they are doing?

**Review by rikimbili — CHANGES_REQUESTED** [review]

File: repos/web/package.json (hunk 2 above)
Comment (rob-3) [reply repos/web/package.json:11]: According to [their README on npm](https://www.npmjs.com/package/svelte-check), they check for:
- Unused CSS
- Svelte A11y hints
- JavaScript/TypeScript compiler errors

I'm not 100% sure what ESLint checks for, but it had a bunch of errors on unused vars in the shadcn stuff, which is why I turned it off. I'm sure there is some overlap.

Comment (rikimbili) [reply repos/web/package.json:11]: I thought I made that rule a warning only. Right now, ESLint is checking for A11y hints and compiler errors. I'm not sure how relevant the unused CSS check is considering that we are using Tailwind and PostCSS already removes unused CSS. And that check is for CSS in <style> tags right?

Comment (rikimbili) [reply repos/web/package.json:11]: Is it maybe not picking up the config?

Comment (rob-3) [reply repos/web/package.json:11]: The errors are of the form
```
web:lint:   7:7  error    '$$Events' is defined but never used  @typescript-eslint/no-unused-vars
```
Yeah, the tsc checks are probably duplicate, so the value of svelte-check would be the A11y stuff (missing alt for images, click accessibility) and catching dead styles in svelte `<style>` blocks.

Comment (rikimbili) [reply repos/web/package.json:11]: is ESLint not checking A11y? On Webstorm ESLint is catching it.

Comment (rob-3) [reply repos/web/package.json:11]: Just tested, it does. Maybe they use svelte-check inside?

**Pull request closed without merging**